prio/server.c

```diff
@@ -83,6 +83,46 @@ PrioServer_aggregate(PrioServer s, PrioVerifier v)
   return MPArray_addmod(s->data_shares, arr, &s->cfg->modulus);
 }
 
+static int
+public_key_cmp(const_PublicKey pub, const_PublicKey pub_other)
+{
+  unsigned char data[CURVE25519_KEY_LEN];
+  unsigned char data_other[CURVE25519_KEY_LEN];
+
+  // if one of the keys is null, both of the keys must be null
+  if (pub == NULL || pub_other == NULL) {
+    return !(pub == NULL && pub_other == NULL);
+  }
+
+  PublicKey_export(pub, data, CURVE25519_KEY_LEN);
+  PublicKey_export(pub_other, data_other, CURVE25519_KEY_LEN);
+  return strncmp(
+    (const char*)data, (const char*)data_other, CURVE25519_KEY_LEN);
+}
+
+static int
+server_cmp(PrioServer s, const_PrioServer s_i)
```

Comment:
One other small aesthetic comment:

This function seems like it's doing two things: (1) comparing `PrioServer` objects and (2) comparing `PrioConfig` objects. It might be more natural to move the config comparison into a public function in `config.c` called `PrioConfig_cmp` that compares configs. Then if we later add extra fields to the Prio config struct, we'll remember to update the comparison operator.

What do you think? If it's easy for you to make that change, I'm happy to review it again before we merge.

Reply:
I agree that seems like the natural way to expose comparison of the config files. Maybe in another PR though, it's the tests that take the most time to write. It should be pretty straightforward including the python bits.

Reply:
Works for me! Merged.